pkgs/coverage/test/test_util.dart

```diff
@@ -89,3 +92,123 @@ extension ListTestExtension on List {
+    branchCoverage: args['branch-coverage'] as bool,
+    packageDirectory: defaultOptions.packageDirectory,
+    packageName: defaultOptions.packageName,
+    testScript: defaultOptions.testScript,
+  );
+}
+
+CoverageOptions parseArgsFormatCoverage(
+    List<String> arguments, CoverageOptions defaultOptions) {
+  final parser = ArgParser()
+    ..addOption('package', defaultsTo: defaultOptions.packageDirectory)
+    ..addOption('out', abbr: 'o');
+
+  final args = parser.parse(arguments);
+
+  String out;
+  final outPath = args['out'] as String?;
+  if (outPath == null && defaultOptions.outputDirectory == null) {
+    out = 'stdout';
+  } else {
+    out = outPath ?? p.join(defaultOptions.outputDirectory!, 'lcov.info');
+  }
+
+  return CoverageOptions(
+    outputDirectory: out,
+    packageDirectory: args['package'] as String,
+    branchCoverage: defaultOptions.branchCoverage,
+    functionCoverage: defaultOptions.functionCoverage,
+    packageName: defaultOptions.packageName,
+    scopeOutput: defaultOptions.scopeOutput,
+    testScript: defaultOptions.testScript,
+  );
+}
+
+Future<CoverageOptions> parseArgsTestWithCoverage(
+    List<String> arguments, CoverageOptions defaultOptions) async {
+  final parser = ArgParser()
+    ..addOption(
+      'package',
+      defaultsTo: defaultOptions.packageDirectory,
+    )
+    ..addOption(
+      'package-name',
+      defaultsTo: defaultOptions.packageName,
+    )
+    ..addOption(
+      'out',
+      defaultsTo: defaultOptions.outputDirectory,
+      abbr: 'o',
+    )
+    ..addOption('test', defaultsTo: defaultOptions.testScript)
+    ..addFlag(
+      'function-coverage',
+      abbr: 'f',
+      defaultsTo: defaultOptions.functionCoverage,
+    )
+    ..addFlag(
+      'branch-coverage',
+      abbr: 'b',
+      defaultsTo: defaultOptions.branchCoverage,
+    )
+    ..addMultiOption('scope-output', defaultsTo: defaultOptions.scopeOutput);
+
+  final args = parser.parse(arguments);
+
+  final packageDir = p.canonicalize(args['package'] as String);
+  if (!FileSystemEntity.isDirectorySync(packageDir)) {
+    ArgumentError('Invalid package directory: $packageDir');
+  }
+
+  final packageName =
+      args['package-name'] ?? await _packageNameFromConfig(packageDir);
+  if (packageName == null) {
+    ArgumentError('Could not determine package name');
+  }
+
+  return CoverageOptions(
+    packageDirectory: packageDir,
+    packageName: packageName as String,
+    outputDirectory: (args['out'] as String?) ?? 'coverage',
+    testScript: args['test'] as String,
+    functionCoverage: args['function-coverage'] as bool,
+    branchCoverage: args['branch-coverage'] as bool,
+    scopeOutput: args['scope-output'] as List<String>,
+  );
+}
+
+Future<String?> _packageNameFromConfig(String packageDir) async {
+  final config = await findPackageConfig(Directory(packageDir));
+  return config?.packageOf(Uri.directory(packageDir))?.name;
+}
+
+String getAbsolutePath(final String package) {
```

Comment:
This function doesn't do much. Your test file can just call `p.canonicalize` directly (it's fine to use canonicalize in tests, even though it's still overpowered, because performance doesn't matter so much in tests).